modules/ml/src/main/java/org/apache/ignite/ml/math/util/MatrixUtil.java

```diff
@@ -209,6 +213,92 @@ public static void unflatten(double[] fArr, Matrix mtx) {
                 mtx.setX(i, j, fArr[!isRowMode ? i * colsCnt + j : j * rowsCnt + i]);
     }
 
+    public static Matrix zipWith(Matrix mtx1, Matrix mtx2, IgniteBiFunction<Double, Double, Double> f) {
+        int rows = Math.min(mtx1.rowSize(), mtx2.rowSize());
+        int cols = Math.min(mtx1.columnSize(), mtx2.columnSize());
+
+        Matrix res = mtx1.like(rows, cols);
+
+        for (int row = 0; row < rows; row++)
+            for (int col = 0; col < cols; col++)
+                res.setX(row, col, f.apply(mtx1.getX(row, col), mtx2.getX(row, col)));
+
+        return res;
+    }
+
+    public static Vector zipWith(Vector v1, Vector v2, IgniteBiFunction<Double, Double, Double> f) {
+        int size = Math.min(v1.size(), v2.size());
+
+        Vector res = v1.like(size);
+
+        for (int row = 0; row < size; row++)
+            res.setX(row, f.apply(v1.getX(row), v2.getX(row)));
+
+        return res;
+    }
+
+    public static Matrix zipWith(Matrix mtx1, Matrix mtx2, IgniteTriFunction<Double, Double, IgniteBiTuple<Integer, Integer>, Double> f) {
```

Comment:
Missed javadoc and explanation.
Example should be added to the JavaDoc due to zip is not popular in Java Frameworks (I see in Scala only)